crates/context/config/src/client/env/config/query/application.rs

```diff
@@ -41,10 +44,45 @@ impl Method<Starknet> for ApplicationRequest {
     const METHOD: &'static str = "application";
 
     fn encode(self) -> eyre::Result<Vec<u8>> {
-        todo!()
+        // Split context_id into high/low parts
+        let bytes = self.context_id.as_bytes();
+        let mid_point = bytes
+            .len()
+            .checked_div(2)
+            .ok_or_else(|| eyre::eyre!("Length should be even"))?;
+        let (high_bytes, low_bytes) = bytes.split_at(mid_point);
+
+        // Convert to Felts
+        let high_felt = Felt::from_bytes_be_slice(high_bytes);
+        let low_felt = Felt::from_bytes_be_slice(low_bytes);
+
+        // Convert both Felts to bytes and concatenate
+        let mut result = Vec::new();
+        result.extend_from_slice(&high_felt.to_bytes_be());
+        result.extend_from_slice(&low_felt.to_bytes_be());
+        Ok(result)
     }
 
-    fn decode(_response: Vec<u8>) -> eyre::Result<Self::Returns> {
-        todo!()
+    fn decode(response: Vec<u8>) -> eyre::Result<Self::Returns> {
+        if response.is_empty() {
+            return Err(eyre::eyre!("No application found"));
+        }
+
+        // Convert bytes to Felts
+        let mut felts = Vec::new();
+        for chunk in response.chunks(32) {
+            if chunk.len() == 32 {
```

Comment:
`chunk_exact`? and if there are any chunks that aren't 32 bytes, what does that mean? is it safe to ignore?